src/lib/MiradorViewer.js

```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { Provider } from 'react-redux';
+import deepmerge from 'deepmerge';
+import ConnectedApp from '../components/App';
+import createRootReducer from '../reducers/index';
+import { store, actions } from '../store';
+import settings from '../config/settings';
+
+/**
+ * Default Mirador instantiation
+ */
+class MiradorViewer {
+  /**
+   */
+  constructor(config) {
+    this.config = config;
+    this.processPlugins();
+
+    const viewer = {
+      actions,
+      store,
+    };
+
+    const action = actions.setConfig(deepmerge(settings, config));
+    store.dispatch(action);
+
+    ReactDOM.render(
+      <Provider store={store}>
+        <ConnectedApp config={config} />
+      </Provider>,
+      document.getElementById(config.id),
+    );
+
+    return viewer;
```

Comment:
Classes inherently return themselves. This class will return something different depending on whether it is called with new or not.

Reply:
Gotcha. Right now we aren't exposing this class publicly so we can control who/where it is instantiated, right?